test/Microsoft.AspNetCore.WebSockets.Server.Test/Autobahn/AutobahnTester.cs

```diff
@@ -0,0 +1,144 @@
+        private readonly List<AutobahnExpectations> _expectations = new List<AutobahnExpectations>();
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger _logger;
+
+        public AutobahnSpec Spec { get; }
+
+        public AutobahnTester(ILoggerFactory loggerFactory, AutobahnSpec baseSpec) : this(7000, loggerFactory, baseSpec) { }
+        public AutobahnTester(int startPort, ILoggerFactory loggerFactory, AutobahnSpec baseSpec)
+        {
+            _nextPort = startPort;
+            _loggerFactory = loggerFactory;
+            _logger = _loggerFactory.CreateLogger("AutobahnTester");
+
+            Spec = baseSpec;
+        }
+
+        public async Task<AutobahnResult> Run()
+        {
+            var specFile = Path.GetTempFileName();
+            try
+            {
+                Spec.WriteJson(specFile);
+
+                // Run the test (write something to the console so people know this will take a while...)
+                _logger.LogInformation("Now launching Autobahn Test Suite. This will take a while.");
+                var result = await Wstest.Default.ExecAsync("-m fuzzingclient -s " + specFile);
+                if (result.ExitCode != 0)
+                {
+                    throw new Exception("wstest failed, stderr: " + result.StdErr);
+                }
+            }
+            finally
+            {
+                if (File.Exists(specFile))
+                {
+                    File.Delete(specFile);
+                }
+            }
+
+            // Parse the output.
+            var outputFile = Path.Combine(Directory.GetCurrentDirectory(), Spec.OutputDirectory, "index.json");
+            using (var reader = new StreamReader(File.OpenRead(outputFile)))
+            {
+                return AutobahnResult.FromReportJson(JObject.Parse(await reader.ReadToEndAsync()));
+            }
+        }
+
+        public void Verify(AutobahnResult result)
+        {
+            var failures = new StringBuilder();
+            foreach(var serverResult in result.Servers)
+            {
+                var serverExpectation = _expectations.FirstOrDefault(e => e.Server == serverResult.Server && e.Ssl == serverResult.Ssl);
+                if(serverExpectation == null)
+                {
+                    failures.AppendLine($"Expected no results for server: {serverResult.Name} but found results!");
+                }
+                else
+                {
+                    serverExpectation.Verify(serverResult, failures);
+                }
+            }
+
+            Assert.True(failures.Length == 0, "Autobahn results did not meet expectations:" + Environment.NewLine + failures.ToString());
+        }
+
+        public async Task DeployTestAndAddToSpec(ServerType server, bool ssl, string environment, Action<AutobahnExpectations> expectationConfig = null)
+        {
+            var port = Interlocked.Increment(ref _nextPort);
```

Comment:
No different than most of our IIS Express-based tests. It'll fail. Today all our IIS Express tests use pre-assigned ports.